### PR TITLE
Migrate 'test_conf.yml' to 'assert_type' tests

### PR DESCRIPTION
**Pull request by flaeppe**
I tried to get started on moving `.yml` tests to `assert_type` style tests. Also tried to adjust the cases to fit the `assert_type` style a little bit better.

Let me know what you think.

### CODE REVIEW

File: django-stubs/urls/conf.pyi [hunk 1]
```diff
@@ -11,9 +11,7 @@ from ..http.response import HttpResponseBase
 
 _URLConf: TypeAlias = str | ModuleType | Sequence[_AnyURL]
 
-def include(
-    arg: _URLConf | tuple[_URLConf, str], namespace: str | None = ...
-) -> tuple[Sequence[URLResolver | URLPattern], str | None, str | None]: ...
+def include(arg: _URLConf | tuple[_URLConf, str], namespace: str | None = ...) -> IncludedURLConf: ...
```
Comment (sobolevn) [inline django-stubs/urls/conf.pyi]: Maybe we can:
- rename it to `_IncludedURLConf`
or
- add it to `ext` module and import from there?

**Review by sobolevn — APPROVED** [review]
Test changes look good! Thanks! One question

File: django-stubs/urls/conf.pyi (hunk 1 above)
Comment (flaeppe) [reply django-stubs/urls/conf.pyi]: Perhaps prefix with an underscore to `_IncludedURLConf`? Thinking since it's only an alias

Comment (flaeppe) [reply django-stubs/urls/conf.pyi]: Done

**Review by sobolevn — APPROVED** [review]

**Pull request merged**